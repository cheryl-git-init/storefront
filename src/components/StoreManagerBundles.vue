<template>
  <div class="bundle-wrapper" >
      <h2>Bundles</h2>
      <p>All bundles give 10% off the items when purchased together!</p>
      <div v-for="bundle in bundles" :key="bundle.name">
          <p class="bundle-name">{{ bundle.name }}:</p>
          <span v-for="(item, index) in bundle.items" :key="item">
              <StoreManagerBundleItem :item="item" /><span v-if="!(index == (bundle.items.length - 1))"> | </span>
          </span>
          <template ></template>
      </div>
  </div>
</template>

<script>
// @ is an alias to /src
import StoreManagerBundleItem from '@/components/StoreManagerBundleItem.vue'


export default {
  name: 'StoreManagerBundles',
  components: {
    StoreManagerBundleItem
  },
  data: function() {
    return {}
  },
  methods: {
  },
  mounted: function() {
  },
  computed: {
    bundles: function () {
      return (this.$store.getters.bundlesList)
    },
  }
}
</script>

<style lang="scss" scoped>
div .bundle-wrapper {
    top: 60px;
    position: relative;
    padding-bottom: 50px;
    display: block !important;
    .bundle-name {
      font-weight: bold;
    }
}
</style>