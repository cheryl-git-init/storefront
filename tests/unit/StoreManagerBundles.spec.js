import Vuex from 'vuex';
import StoreManagerBundles from '@/components/StoreManagerBundles.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('StoreManagerBundles.vue', () => {    
    const wrapper = shallowMount(StoreManagerBundles, {
        computed: { 
            bundles() {
                return [
                    {
                      name: "Brawler Bundle",
                      items: ["7dgX6XzU3Wds", "VqKb4tyj9V6i"]
                    },
                    {
                      name: "Money Bags",
                      items: ["IP3cv7TcZhQn", "500R5EHvNlNB"]
                    },
                    {
                      name: "Fighter",
                      items: ["7Hv0hA2nmci7", "PKM5pGAh9yGm", "7dgX6XzU3Wds"]
                    }
                  ]
            },
        },
        localVue,
    })
    it('displays as expected', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
