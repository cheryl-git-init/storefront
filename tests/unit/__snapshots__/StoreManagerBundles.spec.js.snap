// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreManagerBundles.vue displays as expected 1`] = `
<div class="bundle-wrapper">
  <h2>Bundles</h2>
  <p>All bundles give 10% off the items when purchased together!</p>
  <div>
    <p class="bundle-name">Brawler Bundle:</p> <span><storemanagerbundleitem-stub item="7dgX6XzU3Wds"></storemanagerbundleitem-stub><span> | </span></span><span><storemanagerbundleitem-stub item="VqKb4tyj9V6i"></storemanagerbundleitem-stub><!----></span>
  </div>
  <div>
    <p class="bundle-name">Money Bags:</p> <span><storemanagerbundleitem-stub item="IP3cv7TcZhQn"></storemanagerbundleitem-stub><span> | </span></span><span><storemanagerbundleitem-stub item="500R5EHvNlNB"></storemanagerbundleitem-stub><!----></span>
  </div>
  <div>
    <p class="bundle-name">Fighter:</p> <span><storemanagerbundleitem-stub item="7Hv0hA2nmci7"></storemanagerbundleitem-stub><span> | </span></span><span><storemanagerbundleitem-stub item="PKM5pGAh9yGm"></storemanagerbundleitem-stub><span> | </span></span><span><storemanagerbundleitem-stub item="7dgX6XzU3Wds"></storemanagerbundleitem-stub><!----></span>
  </div>
</div>
`;
